src/components/Navbar.js

```diff
@@ -1,23 +1,34 @@
-import { useEffect, useState } from "react";
+import { Fragment, useEffect, useState } from "react";
 import { Link } from "react-router-dom";
 import { useFilter } from "../context/filter-context";
+import { useAuth } from "../context/auth-context";
 import logo from "../assets/furnishify.png";
+import { logout } from "../services";
 
 export const Navbar = () => {
   const [isOpen, setIsOpen] = useState(false);
   const [query, setQuery] = useState("");
-  const { dispatch } = useFilter();
+  const { dispatch: filterDispatch } = useFilter();
+  const {
+    state: { user },
+    dispatch: authDispatch,
+  } = useAuth();
 
   useEffect(() => {
     const id = setTimeout(() => {
```

Comment:
Nice way of using dispatch and setTimeout together to give the feel of an API fetch!

Reply:
It's not for giving the feel of API fetch, Actually it's to prevent multiple dispatch and dispatch only when used pauses for a while. This is helpful when implementing similar feature with api request so you don't make continuous requests to the api. It's called debouncing

Reply:
Oh nice! Thank you for telling!